src/elements/common/annotator-context/__tests__/withAnnotations.test.tsx

```diff
@@ -73,11 +74,51 @@ describe('elements/common/annotator-context/withAnnotations', () => {
                 const contextProvider = getContextProvider(wrapper);
                 expect(contextProvider.exists()).toBeTruthy();
                 expect((contextProvider.props() as ContextProviderProps).value).toEqual({
+                    activeAnnotationId: null,
                     annotation: expectedAnnotation,
                     action: expectedAction,
                     error: expectedError,
                 });
             },
         );
     });
+
+    describe('handleActiveChange()', () => {
+        test.each`
+            annotationId | expected
+            ${null}      | ${null}
+            ${'123'}     | ${'123'}
+        `('should update activeAnnotationId state to reflect value $annotationId', ({ annotationId, expected }) => {
+            const wrapper = getWrapper();
+
+            (wrapper.instance() as Component & ComponentWithAnnotations).handleActiveChange(annotationId);
+            const contextProvider = getContextProvider(wrapper);
+            expect(contextProvider.exists()).toBeTruthy();
+            expect((contextProvider.props() as ContextProviderProps).value).toEqual({
+                activeAnnotationId: expected,
+            });
+        });
+    });
+
+    describe('handleAnnotatorEvent()', () => {
+        test.each`
+            event                          | numCreateCalled | numActiveChangeCalled
+            ${'annotations_create'}        | ${1}            | ${0}
+            ${'annotations_active_change'} | ${0}            | ${1}
+            ${'foo'}                       | ${0}            | ${0}
+        `(
+            'should call appropriate handler based on event $event',
+            ({ event, numCreateCalled, numActiveChangeCalled }) => {
+                const wrapper = getWrapper();
+                const instance = wrapper.instance() as Component & ComponentWithAnnotations;
+                instance.handleAnnotationCreate = jest.fn();
+                instance.handleActiveChange = jest.fn();
+
+                instance.handleAnnotatorEvent({ event });
+
+                expect((instance.handleAnnotationCreate as jest.Mock).mock.calls.length).toBe(numCreateCalled);
```

Comment:
Is this casting needed? TypeScript should be able to infer it based on the assignment above.

Reply:
Yeah it's not inferred for some reason